targets/mbed/source/jerry_run.cpp

```diff
@@ -17,107 +17,87 @@
 #include <stdlib.h>
 #include <stdio.h>
 
-#include "jerry-core/jerry.h"
+#include "jerry-core/jerry-api.h"
 #include "jerry_extapi.h"
 #include "jerry_run.h"
 
 static const char* fn_sys_loop_name = "sysloop";
 
 int js_entry (const char *source_p, const size_t source_size)
 {
-  const jerry_api_char_t *jerry_src = (const jerry_api_char_t *) source_p;
-  jerry_completion_code_t ret_code = JERRY_COMPLETION_CODE_OK;
-  jerry_flag_t flags = JERRY_FLAG_EMPTY;
-  jerry_api_object_t *err_obj_p = NULL;
-  jerry_api_value_t err_value = jerry_api_create_void_value ();
+  const jerry_char_t *jerry_src = (const jerry_char_t *) source_p;
+  jerry_init_flag_t flags = JERRY_INIT_EMPTY;
+  jerry_value_t ret_value = jerry_create_undefined ();
+  jerry_value_t parsed_code = jerry_parse (jerry_src, source_size, false);
 
   jerry_init (flags);
-
   js_register_functions ();
 
-  if (!jerry_parse (jerry_src, source_size, &err_obj_p))
-  {
-    printf ("Error: jerry_parse failed\r\n");
-    ret_code = JERRY_COMPLETION_CODE_UNHANDLED_EXCEPTION;
-    jerry_api_release_object (err_obj_p);
-  }
-  else
+  if (!jerry_value_has_error_flag (parsed_code))
   {
-    if ((flags & JERRY_FLAG_PARSE_ONLY) == 0)
-    {
-      ret_code = jerry_run (&err_value);
-      jerry_api_string_t *err_str_p = NULL;
-
-      if (err_str_p != NULL)
-      {
-        jerry_api_release_string (err_str_p);
-      }
-    }
-  }
+    ret_value = jerry_run (parsed_code);
 
-  return ret_code;
+    jerry_release_value (ret_value);
+  }
+  
+  return ret_value;
 }
 
 int js_eval (const char *source_p, const size_t source_size)
 {
-  jerry_completion_code_t status;
-  jerry_api_value_t res;
+  jerry_value_t ret_val = jerry_create_undefined ();
 
-  status = jerry_api_eval ((jerry_api_char_t *) source_p,
-                           source_size,
-                           false,
-                           false,
-                           &res);
+  ret_val = jerry_eval ((jerry_char_t *) source_p,
+                        source_size,
+                        ret_val);
 
-  jerry_api_release_value (&res);
+  jerry_release_value (ret_val);
 
-  return status;
+  return 0;
 }
 
 int js_loop (uint32_t ticknow)
 {
-  jerry_api_object_t *global_obj_p;
-  jerry_api_value_t sysloop_func;
-  bool is_ok;
-
-  global_obj_p = jerry_api_get_global ();
-  is_ok = jerry_api_get_object_field_value (global_obj_p,
-                          (const jerry_api_char_t*) fn_sys_loop_name,
-                          &sysloop_func);
-  if (!is_ok)
+  jerry_value_t global_obj_p;
+  jerry_value_t sysloop_func = jerry_create_undefined ();
+  jerry_value_t prop_value;
+
+  global_obj_p = jerry_get_global_object ();  
+  jerry_value_t prop_value_name_val = jerry_create_string ((const jerry_char_t*) fn_sys_loop_name);
+  prop_value = jerry_get_property (global_obj_p, prop_value_name_val);
```

Comment:
sysloop_func is unused, you can use it instead of prop_value and remove `prop_value`